root/static/scripts/edit/URLCleanup.js

```diff
@@ -1816,8 +1816,13 @@ const CLEANUPS = {
     },
   },
   'secondhandsongs': {
-    match: [new RegExp('^(https?://)?([^/]+\\.)?secondhandsongs\\.com/', 'i')],
+    match: [new RegExp('^(https?://)?(?:[^/]+\\.)?secondhandsongs\\.com/', 'i')],
     type: LINK_TYPES.secondhandsongs,
+    clean: function (url) {
+      url = url.replace(/^(?:https?:\/\/)?(?:[^/]+\.)?secondhandsongs\.com/, 'https://secondhandsongs.com');
+      url = url.replace(/^https:\/\/secondhandsongs\.com\/(\w+\/[\d+]+)[\/#?-].*$/, 'https://secondhandsongs.com/$1');
```

Comment:
Capturing group could start with `https`… so that replacement string just needs to be `'$1'`.